Sources/Fuzzilli/FuzzIL/TypeSystem.swift

```diff
@@ -850,53 +789,168 @@ class TypeExtension: Hashable {
     }
 }
 
+public enum Parameter: Equatable, Hashable {
+    case plain(Type)
+    case opt(Type)
+    case rest(Type)
+
+    fileprivate func format(abbreviate: Bool) -> String {
+        switch self {
+            case .plain(let t):
+                return ".plain(\(t.format(abbreviate: abbreviate)))"
+            case .opt(let t):
+                return ".opt(\(t.format(abbreviate: abbreviate)))"
+            case .rest(let t):
+               return ".rest(\(t.format(abbreviate: abbreviate)))"
+        }
+    }
+
+    fileprivate func callerType() -> Type {
+        switch self {
+            case .plain(let t):
+                return t
+            case .opt(let t):
+                return t
+            case .rest(_):
+               return .iterable
+        }
+    }
+
+    fileprivate func calleeType() -> [Type] {
+        return [self.callerType()]
```

Comment:
I think here optionals should get the union with `.undefined` and rest should get `.iterable`?

Reply:
See comment in `AbstractInterpreter`. I think we should just move this method into the AI entirely, where we e.g. have access to the correct ArrayType to use for rest parameters

Reply:
> I think here optionals should get the union with .undefined and rest should get .iterable?

I tried this before and I think I ran into a number of assertion failures when merging types. I'll give it another shot and report back.

Reply:
I'm at 200k iterations and I haven't seen any assertion failures yet. I think this is good to go!